manual/src/tutorials/moduleexamples.etex

```diff
@@ -130,50 +114,54 @@ Signatures are interfaces for structures. A signature specifies
 which components of a structure are accessible from the outside, and
 with which type. It can be used to hide some components of a structure
 (e.g. local function definitions) or export some components with a
-restricted type. For instance, the signature below specifies the three
-priority queue operations "empty", "insert" and "extract", but not the
-auxiliary function "remove_top". Similarly, it makes the "queue" type
+restricted type. For instance, the signature below specifies the
+queue operations "empty", "add", "top" and "pop", but not the
+auxiliary function "make". Similarly, it makes the "queue" type
 abstract (by not providing its actual representation as a concrete type).
+This ensures that users of the "Fifo" module cannot violate data
+structure invariants that operations rely on, such as ``if the front list
+is empty, the rear list is empty too''.
 \begin{caml_example}{toplevel}
-module type PRIOQUEUE =
+module type FIFO =
   sig
-    type priority = int         (* still concrete *)
-    type 'a queue               (* now abstract *)
+    type 'a queue               (* now an abstract type *)
     val empty : 'a queue
-    val insert : 'a queue -> int -> 'a -> 'a queue
-    val extract : 'a queue -> int * 'a * 'a queue
-    exception Queue_is_empty
+    val add : 'a -> 'a queue -> 'a queue
+    val top : 'a queue -> 'a
+    val pop : 'a queue -> 'a queue
+    exception Empty
   end;;
 \end{caml_example}
-Restricting the "PrioQueue" structure by this signature results in
-another view of the "PrioQueue" structure where the "remove_top"
-function is not accessible and the actual representation of priority
+Restricting the "Fifo" structure by this signature results in
+another view of the "Fifo" structure where the "make"
+function is not accessible and the actual representation of
 queues is hidden:
 \begin{caml_example}{toplevel}
-module AbstractPrioQueue = (PrioQueue : PRIOQUEUE);;
-AbstractPrioQueue.remove_top [@@expect error];;
-AbstractPrioQueue.insert AbstractPrioQueue.empty 1 "hello";;
+module AbstractQueue = (Fifo : FIFO);;
+AbstractQueue.make [1] [2;3] [@@expect error];;
+AbstractQueue.add "hello" AbstractQueue.empty;;
 \end{caml_example}
 The restriction can also be performed during the definition of the
 structure, as in
 \begin{verbatim}
-module PrioQueue = (struct ... end : PRIOQUEUE);;
+module Fifo = (struct ... end : FIFO);;
 \end{verbatim}
 An alternate syntax is provided for the above:
 \begin{verbatim}
-module PrioQueue : PRIOQUEUE = struct ... end;;
+module Fifo : FIFO = struct ... end;;
 \end{verbatim}
 
 Like for modules, it is possible to include a signature to copy
 its components inside the current signature. For instance, we
```

Comment:
Not touched by this PR, but for clarity, I would rewrite as follows (the reference to the "current" siganture is not very clear):
```suggestion
As for modules, it is possible to copy the components of a signature inside another signature by using an "include" statement. For instance, we
```

Reply:
Agreed with the clarification, but "as for modules" means "as regards modules" or "as far as modules are concerned", so it should be "as with modules".

Reply:
I agree.